src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -418,6 +418,25 @@ impl ScalarFunc {
     ) -> Result<Option<Cow<'a, MyDuration>>> {
         Ok(Some(Cow::Owned(MyDuration::zero())))
     }
+
+    #[inline]
+    pub fn sub_duration_and_duration<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, MyDuration>>> {
+        let d0: Cow<'a, MyDuration> = try_opt!(self.children[0].eval_duration(ctx, row));
+        let d1: Cow<'a, MyDuration> = try_opt!(self.children[1].eval_duration(ctx, row));
+        let diff = match d0.to_nanos().checked_sub(d1.to_nanos()) {
+            Some(result) => result,
+            None => return Err(box_err!("sub duration {} from duration {} error", d1, d0)),
```

Comment:
Seems we should return an error with code `ERR_DATA_OUT_OF_RANGE`

Reply:
@AndreMouche PTAL.